src/main/java/org/cloudburstmc/server/config/ServerProperties.java

```diff
@@ -109,17 +109,17 @@ public Path getPath() {
     public void setPath(Path path) { this.path = path; }
 
 
-    public void setWhitelist(boolean b) {
+    public void modifyWhitelist(boolean b) {
         this.whiteList = b;
```

Comment:
Sorry about the constant reviews, but do you think it would be better to have to explicitly call `save()` instead of it being called automatically? 

That’s a question, I’m not suggesting anything as I’m not sure myself. If you do choose to do that though, you should probably change it back to setWhitelist

Reply:
It is great, I should be the one to sorry for constantly bothering you :)

I am also unsure about this. Places originally setting these properties demand persistence, I think I can pull the save() into the facade, which could make these modifiers go away. Further pulling save() into call site might be a bad idea tho? Because we would then see commands like /difficulty trying to save a config file, which might be undesirable?

tbh I think it is a bad idea to modify config file at run time, the setter should just set the state in memory and the server owner should modify the config file themselves if they want their config to persist across reboot.

Reply:
While it would be nice, that's just an inconvenience to the server owner. In vanilla, executing the difficulty command will change the difficulty in the `level.dat` and save it automatically, so it should be expected to change in the `server.properties` as well, i guess.

My question was more about the API and whether we should allow plugins to change a setting without saving it straight away (such as if they want to change multiple settings). This is probably a discussion for another time and another PR, though. Ideally, we would not allow plugins to modify the server config via the API (if a plugin really needed to modify it they could still do it via other means), but i don't think you need to deal with that in your PR.